core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -761,16 +761,20 @@ CompressionCodec initCodec(final AtomicBoolean checked, final int bufferSize,
      */
     CompressionCodec createNewCodec(final String codecClazzProp, final String defaultClazz,
         final int bufferSize, final String bufferSizeConfigOpt) {
-      String extClazz =
-          (conf.get(codecClazzProp) == null ? System.getProperty(codecClazzProp) : null);
+      String extClazz = conf.get(codecClazzProp);
+      if (extClazz == null) {
+        extClazz = System.getProperty(codecClazzProp);
+      }
       String clazz = (extClazz != null) ? extClazz : defaultClazz;
```

Comment:
The behavior comes from the existing Accumulo code which didn't work and the Hadoop TFile example linked here: https://github.com/apache/hadoop/blob/e103c83765898f756f88c27b2243c8dd3098a989/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/file/tfile/Compression.java#L88 - It is a bit backwards I agree and may be more Hadoop semantics rather than Accumulo.  I would agree with your approach.

Reply:
Yeah, the simpler version of that implementation would look like:

```java
      String clazz = conf.get(codecClazzProp, System.getProperty(codecClazzProp, defaultClazz));
```

But, that would make the value in the config file override a java property on the command-line. And, I think it should go the other way around. The command-line is should take precedence.